### PR TITLE
249 feat qf calculations

### PR DESCRIPTION
**Pull request by hussmartinez**
- implemented round qf distribution calculations -> `lib/qf-calculation`

UI Changes
- changed MatchingFundsHighlight to display actual matching data
  - CampaignRoundMatching uses the hook useCampaignMatching that fetches the round distribution and return the campaignMatchingAmount
  
 It still needs some polishing in terms of having hardcoded things like token decimals.
  
 Loom with a bit of clarification -> https://www.loom.com/share/9366f619a4b24dbda66df9f79b1fc045?sid=9fc49951-c8cd-4596-81ce-83b3ad34de77

## Summary by CodeRabbit

- New Features
  - Display estimated matching amounts for campaign rounds, with clear “pending” and “TBD” states.
  - Show matching pool totals alongside round titles.
  - Automatically determines round status (Active, Upcoming, Ended) and only fetches estimates when relevant.
  - Improved accuracy of matching distribution with precise rounding and dust handling.

- Tests
  - Added comprehensive tests for parsing round data and calculating quadratic funding distributions, covering edge cases and precision scenarios.

### CODE REVIEW

**Review by joerg-oss — APPROVED** [review]
Great Work!

I like that you added tests and applied the new patterns.

comments are for considerations and putting new tickets on the roadmap.

File: app/api/qf-calculation/[id]/route.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { response, handleError } from '@/lib/api/response';
+import { RoundsWithIdParams } from '@/lib/api/types';
+import { getRoundQFDistribution } from '@/lib/qf-calculation';
+import { debugApi as debug } from '@/lib/debug';
+
+export async function GET(req: Request, { params }: RoundsWithIdParams) {
+  const id = parseInt((await params).id);
+  try {
+    debug && console.log(`Starting QF calculation for round ${id}...`);
+    const { distribution, totalAllocated } = await getRoundQFDistribution(id);
```
[the remark this answers is not in the document]
Comment (hussmartinez) [reply app/api/qf-calculation/[id]/route.ts:10]: Definitely, we should implement cache for this,
Imo invalidating only when a new round contribution is done and after the round ends and is distributed then we could store distribution in db to avoid new calculations when we show history
